### PR TITLE
Improve js target of Haxe compiler ...

### PR DESCRIPTION
**Pull request by bjitivo**
This makes the JavaScript back-end stream the generated JavaScript to the
output file instead of trying to build it whole-cloth in RAM first.

### CODE REVIEW

**Comment by ncannasse** [comment]
Could the flush not be done after each class or method instead ? should be more efficient

**Comment by Simn** [comment]
Ooh this must be that PR Todd has wanted to send me for two years! :D

**Comment by bjitivo** [comment]
Yes it's Todd's change, so this is probably the one.  I'll ask him if he agrees that the flush would be better after every class or method.  Hopefully I'll have an answer for that sooner than 2 more years ...

**Comment by bjitivo** [comment]
Updated as per your request.  Please pull.  Thanks!

**Comment by nadako** [comment]
Could you maybe squash your PR in one commit? Because it looks very weird to have 20+ commits for such a small change. Also there's some unrelated whitespace change in `extra/release-checklist.txt`.

**Comment by Simn** [comment]
Can't merge this, I'll deal with it locally.

File: genjs.ml [hunk 1]
```diff
@@ -1129,7 +1136,8 @@ let generate_enum ctx e =
 	| Some e ->
 		print ctx "%s.__meta__ = " p;
 		gen_expr ctx e;
-		newline ctx
+		newline ctx;
+	flush ctx
```
Comment (Simn) [inline genjs.ml]: Note that this is part of the `Some e` branch which the indent suggests it shouldn't be.

**Pull request closed without merging**